### PR TITLE
Added support to subscribe existing log groups by User defined Tags

### PR DESCRIPTION
**Pull request by akhil-sumologic**

### CODE REVIEW

File: loggroup-lambda-connector/src/loggroup-lambda-connector.js [hunk 1]
```diff
@@ -1,11 +1,82 @@
...
 const cwl = new CloudWatchLogsClient();
 const lambda = new LambdaClient({ apiVersion: '2015-03-31' }); // Update to the appropriate Lambda API version you require
 const maxRetryCounter = 3;
 const timeoutThreshold = 12000;
 
+
+function validateRegex(pattern) {
+    try {
+        // Attempt to create a RegExp object with the provided pattern
+        return new RegExp(pattern, "i");
```
[the remark this answers is not in the document]
Comment (akhil-sumologic) [reply loggroup-lambda-connector/src/loggroup-lambda-connector.js:13]: I attempted to fix this, but it ended up changing the function's behaviour.

**Review by himsharma01 — APPROVED** [review]
lgtm

**Review by himsharma01 — CHANGES_REQUESTED** [review]
minor copy text commit

File: loggroup-lambda-connector/Readme.md [hunk 2]
```diff
@@ -25,13 +25,30 @@ Made with ❤️ by Sumo Logic. Available on the [AWS Serverless Application Rep
...
+| To subscribe all loggroup                                            | `/*` or (leave empty)               |
+| To subscribe all loggroup paths only                                 | `/`                                 |
+| To subscribe all loggroup of aws services                            | `/aws/*`                            |
+| To subscribe to loggroups for only one service, such as Lambda       | `/aws/lambda/*`                     |
+| To subscribe loggroup multiple services like lambda, rds, apigateway | `/aws/(lambda\|rds\|apigateway)`    |
+| To subscribe loggroup by key word like `Test` or `Prod`              | `Test` or `Prod` [Case insensitive] |
+| Don't subscribe if `LOG_GROUP_PATTERN`                               | `^$`                                |
+
+**LOG_GROUP_TAGS**: This is used to filter log groups based on tags. Only log groups that match any of the specified key-value pairs will be subscribed to the Lambda function. It is case-sensitive.
+#### e.g
```
[the remark this answers is not in the document]
Comment (akhil-sumologic) [reply loggroup-lambda-connector/Readme.md]: Done.

**Review by himsharma01 — APPROVED** [review]

**Review by himsharma01 — APPROVED** [review]
lgtm

**Pull request merged**